### PR TITLE
Document schedule data property

### PR DESCRIPTION
**Pull request by Klikini**
## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/116585

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
  - Enhanced YAML schedule configuration by allowing additional data attributes for schedule blocks, enabling richer context for automations.
  - Introduced the `data` key for defining specific attributes like "brightness" and "color_temp" within schedule blocks.
- **Documentation**
  - Updated documentation to include details about the new `data` key, its optional nature, and practical examples of usage.

### CODE REVIEW

File: source/_integrations/schedule.markdown [hunk 1]
```diff
@@ -63,6 +63,66 @@ schedule:
...
+      - from: "07:00:00"
+        to: "10:00:00"
+        data:
+          brightness: 80
+          color_temp: 3000
+      - from: "16:00:00"
+        to: "23:00:00"
+        data:
+          brightness: 60
+          color_temp: 2500
```
[the remark this answers is not in the document]
Comment (Klikini) [reply source/_integrations/schedule.markdown]: I think this is self-explanatory to humans, but apparently not to robots. I'd be happy to provide example automations if a real person thinks it's worth cluttering the schedule page with them.

File: source/_integrations/schedule.markdown [hunk 2]
```diff
@@ -63,6 +63,66 @@ schedule:
         to: "21:00:00"
 ```
 
+Defining the schedule in YAML also allows adding extra data to each block, which will
+appear as attributes on the schedule helper entity when that block is active, and can
+be used to easily build schedule-based automations.
```
[the remark this answers is not in the document]
Comment (Klikini) [reply source/_integrations/schedule.markdown]: It's already clearly listed out next to the data attribute under "configuration variables"...can this overgrown linear regression not parse that part?

**Comment by Klikini** [comment]
There were some last-minute changes to the parent PR that we need to account for here before this gets merged. I can handle that later tonight.

**Comment by Klikini** [comment]
Okay, once those builds finish, it should be good to go.

**Review by c0ffeeca7 — DISMISSED** [review]
Thank you, @Klikini :+1:

**Comment by c0ffeeca7** [comment]
A merge conflict emerged. Could you take a look?

**Comment by c0ffeeca7** [comment]
I'm [drafting this PR](https://developers.home-assistant.io/docs/review-process#prs-are-being-drafted-when-changes-are-needed), as changes are needed. Please undraft it, once the changes have been implemented and the PR is ready for review again.

**Review by c0ffeeca7 — APPROVED** [review]

**Pull request merged**